libwaysip/src/state.rs

```diff
@@ -0,0 +1,315 @@
+
+
+use wayland_client::{
+    protocol::{
+        wl_buffer::WlBuffer,
+        wl_surface::WlSurface,
+    },
+};
+
+
+
+use wayland_protocols::xdg::xdg_output::zv1::client::{
+    zxdg_output_v1,
+};
+
+use wayland_protocols_wlr::layer_shell::v1::client::{
+    zwlr_layer_surface_v1::{ZwlrLayerSurfaceV1},
+};
+
+use wayland_protocols::wp::cursor_shape::v1::client::{
+    wp_cursor_shape_manager_v1::WpCursorShapeManagerV1,
+};
+
+use wayland_cursor::CursorImageBuffer;
+
+
+use wayland_client::protocol::wl_output::WlOutput;
+
+/// You are allow to choose three actions of waysip, include area selection, point selection, and
+/// select sreen
+#[derive(Debug, Clone, Copy, Default)]
+pub enum WaySipKind {
```

Comment:
I think this should be pub used at lib.rs

Reply:
It's already pub right? I don't get what you mean

Reply:
`use libwaysip::{get_area, WaySipKind};`

I mean we can use just from libwaysip. yes, it is in state mod, but maybe it should add a 

```rust
pub use state::WaysipKind;
```
in lib.rs

this is my opinion

Reply:
Okay on it.